File: zhanglianzhuang/day2/demo02/src/main/java/com/example/demo02/dao/IUser.java
package com.example.demo02.dao;
import com.example.demo02.entity.User;
import org.apache.ibatis.annotations.Mapper;
import java.util.List;

@Mapper
public interface IUser {
	int insertUser(User user);
	List<User> getUser(User user);
	int updateUser(User user);
	int deleteUser(User user);
}

File: caiguicheng/day03/work/demo03work/dao/IUser.kt
package com.example.demo03work.dao

import com.example.demo03work.entity.User
import org.apache.ibatis.annotations.Mapper

/**
 * 用户操作接口类
 */
@Mapper
interface IUser {

    /**
     * 插入用户
     * @return 是否成功
     */
    fun insertUser(user: User): Int

    /**
     * 更新用户信息
     * @return 是否成功
     */
    fun updateUser(user: User): Int

    /**
     * 根据条件获得用户
     * @return 用户列表
     */
    fun selectUser(user:User):List<User>

    /**
     * 删除用户
     * @return 是否成功
     */
    fun deleteUser(id:Int):Int
}